### PR TITLE
[ts] Enforce order for the `override` modifier

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Ref: https://github.com/microsoft/TypeScript/pull/43660
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 1]
```diff
@@ -267,26 +273,29 @@ export default (superClass: Class<Parser>): Class<Parser> =>
...
+              (modified.declare && modifier === "override") ||
+              (modified.override && modifier === "declare")
+            ) {
+              this.raise(
+                startPos,
+                TSErrors.IncompatibleModifiers,
+                "declare",
+                "override",
+              );
+            }
```
Comment (JLHwung) [inline packages/babel-parser/src/plugins/typescript/index.js:298]: `static` can not be used with `abstract`: [TS Playground](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210426&ssl=1&ssc=1&pln=8&pc=1#code/IYIwzgLgTsDGEAJYBthjAgQmgpgYVXQQG8AoBBSYCAS1gQFsBGALgWADsBPUgX1NKhIMeEkIYAslwJowAVgQ4AHhBwcAJhmxh84kuUoRqdduGhxEzNsJocA5gG4+pIA).

**Review by JLHwung — APPROVED** [review]
Did this PR also introduce `static override` support?

**Comment by nicolo-ribaudo** [comment]
I need to update this PR for https://github.com/microsoft/TypeScript/pull/43829

> Did this PR also introduce static override support?

Yes :+1:

File: packages/babel-parser/src/plugins/typescript/index.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply packages/babel-parser/src/plugins/typescript/index.js:298]: Oh it looks like we didn't have any test for it

**Comment by nicolo-ribaudo** [comment]
CI caught a regression _closely related_ to a TS bug I just found: https://github.com/microsoft/TypeScript/issues/43836. I will only fix the regression partially, and wait for the TS bug to be fixed to understand what is the expected behavior.

**Comment by nicolo-ribaudo** [comment]
Oh it's not a regression, it's a bugfix lol

**Comment by nicolo-ribaudo** [comment]
The new tests in the allowlist are not related to this PR, they are just new TS tests.

**Review by existentialism — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]

**Review by kaicataldo — APPROVED** [review]

**Pull request merged**